src/Provider/Keycloak.php

```diff
@@ -68,11 +83,67 @@ public function __construct(array $options = [], array $collaborators = [])
     }
 
     /**
+     * We need to cache the access token locally allowing for later optional post-processing
+     * by `checkForKeycloakRoles()`
+     *
+     * @param mixed $grant
+     * @param array $options
+     * @return AccessToken
+     */
+    public function getAccessToken($grant, array $options = [])
+    {
+        $this->accessToken = parent::getAccessToken($grant, $options);
+        return $this->accessToken;
+    }
+
+    /**
+     * Check for Keycloak-supplied additional fields held by the access token which in turn is inside accessToken.
+     *
+     * @return KeyCloakRoles
+     */
+    public function getKeycloakRoles()
```

Comment:
The access token workflow leads directly to the unpacking of Keycloak roles yet my changes allow the latter as an optional extra - I'm not seeing any unnecessary coupling, just the cost of maintaining access to the token.

The Keycloak client holds the key and algorithm already and thus can unpack the payload just as it does for the access token. These changes mean clients do not need to pass around keys and algorithms to get access to this functionality post-construction - they are encapsulated in the Keycloak client instance. This has the obvious benefit of maintaining a naive client in my case.

I do not recommend static methods to access these extensions, but feel free to add weight to your argument and I shall reconsider. Of course, an alternate approach would be to create an extension of this project (and thus class) but I'm not drawn to this when these changes are relatively trivial.